source/FreeRTOS_Sockets.c

```diff
@@ -1987,7 +1987,10 @@ BaseType_t FreeRTOS_setsockopt( Socket_t xSocket,
                         /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-111 */
                         /* coverity[misra_c_2012_rule_11_8_violation] */
                         /* coverity[misra_c_2012_rule_11_1_violation] */
+                        #pragma GCC diagnostic push
+                        #pragma GCC diagnostic ignored "-Wpedantic"
                         pxSocket->pxUserWakeCallback = ( SocketWakeupCallback_t ) pvOptionValue;
+                        #pragma GCC diagnostic pop
```

Comment:
Not sure I'm a big fan of this but I had to do this to resolve the final pedantic warning which was around this cast.

I'm open to removing this and disabling pedantic warnings.

Reply:
Wait but it looks like the wpedantic option was removed in the CMakeLists.txt, is it still hitting this?
And I know other compilers will flag pragmas as warnings? Where feel like this might just be trading this warning for another one?

Reply:
TI ARM CGT also complains about this. Can we provide a generic macro for this in ``source/portable/Compiler``? Otherwise one can not consume the common files without changing such code pieces.

I will add a separate issue to discuss this.